Dockerfile

```diff
@@ -1,12 +1,27 @@
-FROM adoptopenjdk:11-jre-hotspot
-
-RUN apt-get update \
-    && apt-get install -y unzip \
-    && rm -rf /var/lib/apt/lists/*
+ARG VERSION=3.6
+ARG JAVA_VERSION=11-jre-hotspot
 
+FROM adoptopenjdk:${JAVA_VERSION}
 ARG VERSION
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
```

Comment:
Why is `-c` used here?

Reply:
This `SHELL` is sanity checking, if we're going to fail let's fail hard. See https://github.com/docker/for-linux/issues/408

Reply:
`-o pipefail` returns the first non-zero exit code in a pipe, but what about `-c`. The manual says

> -c
>    Read and execute commands from the first non-option argument command_string, then exit. If there are arguments after the command_string, the first argument is assigned to $0 and any remaining arguments are assigned to the positional parameters. The assignment to $0 sets the name of the shell, which is used in warning and error messages.

I have trouble parsing that. Could you elaborate what change it introduces?

Reply:
Well you ain't going to need `-c` until you need it. It really only saves you from some weird edge cases. It helps with properly unquoting the command line parameters, say you want to invoke the container passing in the transtype which has been read in using some elaborate bash conversion function or something.

The best explanation I could find was here: https://ask.metafilter.com/316796/Using-piping-with-nix-find-without-getting-mired-in-string-escaping

Reply:
A concrete example would be that now you can run

```bash
$ docker run docker.pkg.github.com/dita-ot/dita-ot/dita-ot:3.6 install foo
```

and with the `-c` shell you can both

```
```bash
$ docker run docker.pkg.github.com/dita-ot/dita-ot/dita-ot:3.6 install foo
$ docker run docker.pkg.github.com/dita-ot/dita-ot/dita-ot:3.6 "install foo"
```

Seems useful.